packages/mobile/src/identity/contactMapping.ts

```diff
@@ -216,15 +221,43 @@ export function* fetchAddressesAndValidateSaga({
   }
 }
 
-function* getAddresses(e164Number: string) {
-  let phoneHash: string
+function* getAccountAddresses(e164Number: string) {
   const phoneHashDetails: PhoneNumberHashDetails = yield call(fetchPhoneHashPrivate, e164Number)
-  phoneHash = phoneHashDetails.phoneHash
-
+  const phoneHash = phoneHashDetails.phoneHash
   const lookupResult: IdentifierLookupResult = yield call(lookupAttestationIdentifiers, [phoneHash])
   return getAddressesFromLookupResult(lookupResult, phoneHash) || []
 }
 
+function* fetchWalletAddresses(e164Number: string) {
+  const contractKit = yield call(getContractKit)
+  const accountsWrapper: AccountsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAccounts,
+  ])
+
+  const accountAddresses: string[] = yield call(getAccountAddresses, e164Number)
+  const walletAddresses: string[] = yield all(
+    accountAddresses.map((accountAddress) => call(accountsWrapper.getWalletAddress, accountAddress))
+  )
+
+  const possibleUserAddresses: string[] = []
+  const walletToAccountAddress: WalletToAccountAddressType = {}
+  walletAddresses.forEach((walletAddress, i) => {
+    const accountAddress = accountAddresses[i]
+    // `getWalletAddress` returns a null address when there isn't a wallet registered
+    if (!eqAddress(walletAddress, NULL_ADDRESS)) {
+      walletToAccountAddress[walletAddress] = accountAddress
+      possibleUserAddresses.push(walletAddress)
+    } else {
+      // NOTE: Only need this else block if we are not confident all wallets are registered
+      walletToAccountAddress[accountAddress] = accountAddress
+      possibleUserAddresses.push(accountAddress)
+    }
+  })
```

Comment:
`.forEach` is a bit of an anti-pattern. Using `for ... of` with `.entries` should be preferred.
```suggestion
 for (const [i, walletAddress] of walletAddresses.entries()) {
    const accountAddress = accountAddresses[i]
    // `getWalletAddress` returns a null address when there isn't a wallet registered
    if (!eqAddress(walletAddress, NULL_ADDRESS)) {
      walletToAccountAddress[walletAddress] = accountAddress
      possibleUserAddresses.push(walletAddress)
    } else {
      // NOTE: Only need this else block if we are not confident all wallets are registered
      walletToAccountAddress[accountAddress] = accountAddress
      possibleUserAddresses.push(accountAddress)
    }
  }
```